src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -225,14 +236,39 @@ private void RemovePlaceholderScope(BoundValuePlaceholderBase placeholder)
 
         private uint GetPlaceholderScope(BoundValuePlaceholderBase placeholder)
         {
-            // https://github.com/dotnet/roslyn/issues/67070: Assert should be re-enabled when placeholders are included.
-            //Debug.Assert(_placeholderScopes?.ContainsKey(placeholder) == true);
+            Debug.Assert(_placeholderScopes?.ContainsKey(placeholder) == true);
 
             return _placeholderScopes?.TryGetValue(placeholder, out var scope) == true
                 ? scope
                 : CallingMethodScope;
         }
 
+#if DEBUG
+        private bool ContainsPlaceholderScope(BoundValuePlaceholderBase placeholder)
+        {
+            if (_placeholderScopes?.ContainsKey(placeholder) == true)
+            {
+                return true;
+            }
+
+            // _placeholderScopes should contain all placeholders that may be used by GetValEscape() or CheckValEscape().
+            // The following placeholders are not needed by those methods and can be ignored, however.
+            switch (placeholder)
+            {
+                case BoundObjectOrCollectionValuePlaceholder:
+                    return true; // CheckValEscapeOfObjectInitializer() does not use BoundObjectOrCollectionValuePlaceholder.
```

Comment:
>// CheckValEscapeOfObjectInitializer() does not use BoundObjectOrCollectionValuePlaceholder. [](http://example.com/codeflow?start=33&length=92)

We don't visit member initializers?

Reply:
We visit member initializers here in the call to `Visit(node.InitializerExpressionOpt)` in `VisitObjectCreationExpressionBase()`, but we don't add the `Placeholder` to the `_placeholders` collection, we simply visit the `Initializers`. And, as mentioned in the comment, `CheckValEscapeOfObjectInitializer()` does not use the escape scope of `Placeholder`.